examples/tools.py

```diff
@@ -63,18 +63,14 @@ def subtract_two_numbers(a: int, b: int) -> int:
       print('Arguments:', tool.function.arguments)
       output = function_to_call(**tool.function.arguments)
       print('Function output:', output)
+      messages.append(response.message.model_dump())
+      messages.append({'role': 'tool', 'content': str(output), 'name': tool.function.name})
     else:
       print('Function', tool.function.name, 'not found')
 
-# Only needed to chat with the model using the tool call results
-if response.message.tool_calls:
-  # Add the function response to messages for the model to use
-  messages.append(response.message)
-  messages.append({'role': 'tool', 'content': str(output), 'name': tool.function.name})
```

Comment:
one issue: `output` is not visible in this scope

Reply:
as well as `tool`

Reply:
Actually, I realize a larger issue around multiple tools getting generated, going to update the structure a bit. Aside from that the positive run case should be working fine.

Reply:
you are right, I am coming from Java universe and fell for for the "leaking" loop variable issue here. Sorry about this 😮‍💨